### PR TITLE
Reporting Interval as property in CustomResourceDefinition

### PR DESCRIPTION
**Pull request by Pejdzor**

### CODE REVIEW

File: api/v1alpha1/ibmlicensing_types.go [hunk 1]
```diff
@@ -186,6 +186,11 @@ type IBMLicensingSenderSpec struct {
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Cluster ID",xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
 	// +optional
 	ClusterID string `json:"clusterID,omitempty"`
+
+	// Frequency of workloads reporting as cron expression. If not provided, workloads reporting is disabled.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Workloads Reporting Frequency",xDescriptors="urn:alm:descriptor:com.tectonic.ui:text"
+	// +optional
+	WorkloadsReportingInterval string `json:"workloadsReportingInterval,omitempty"`
```
Comment (kflorianski-ibm) [inline api/v1alpha1/ibmlicensing_types.go:193]: Please consider adding pattern validation. For k8s cron, this is what I added in the scanner operator:
```
// Frequency of the scans in the cron job format
// +kubebuilder:validation:Pattern:=`(@(annually|yearly|monthly|weekly|daily|midnight|hourly))|((((\d+,)+\d+|(\d+(\/|-)\d+)|\d+|\*) ?){5,7})`
// +optional
Frequency string `json:"frequency,omitempty"`
```

Comment (kflorianski-ibm) [inline api/v1alpha1/ibmlicensing_types.go:193]: The name is a bit lengthy (clients must type `workloadsReportingInterval`) but I don't have a better idea for now, so this comment is just "maybe a random idea will appear"

File: api/v1alpha1/helper.go [hunk 2]
```diff
@@ -193,6 +193,10 @@ func (spec *IBMLicensingSpec) IsChargebackEnabled() bool {
 	return spec.ChargebackEnabled != nil && *spec.ChargebackEnabled
 }
 
+func (spec *IBMLicensingSpec) IsWorkloadsReportingEnabled() bool {
+	return spec.Sender.WorkloadsReportingInterval != ""
+}
+
```
Comment (kflorianski-ibm) [inline api/v1alpha1/helper.go:199]: Unused, you use `spec.Sender.WorkloadsReportingInterval != ""` anyway (which is fine!). Remove the function plz

File: controllers/resources/service/containers.go [hunk 3]
```diff
@@ -184,6 +184,19 @@ func getLicensingEnvironmentVariables(spec operatorv1alpha1.IBMLicensingSpec) []
...
 
+		if spec.Sender.WorkloadsReportingInterval != "" {
+			environmentVariables = append(environmentVariables, []corev1.EnvVar{
+				{
+					Name:  "SENDER_WORKLOADS_INTERVAL",
+					Value: spec.Sender.WorkloadsReportingInterval,
+				},
+				{
+					Name:  "ENABLE_WORKLOADS_REPORTING",
+					Value: "true",
```
Comment (kflorianski-ibm) [inline controllers/resources/service/containers.go:195]: Do we need both? Can't we have a check on the operand side if `SENDER_WORKLOADS_INTERVAL` empty then -> disabled (similarly to how the API value is being described - if not provided, the feature is disabled)

File: controllers/resources/service/containers.go [hunk 4]
```diff
@@ -184,6 +184,19 @@ func getLicensingEnvironmentVariables(spec operatorv1alpha1.IBMLicensingSpec) []
 			}...)
 		}
 
+		if spec.Sender.WorkloadsReportingInterval != "" {
```
Comment (kflorianski-ibm) [inline controllers/resources/service/containers.go:187]: Can you add a comment please -> "If workloads reporting feature is enabled, pass the env vars marking this to the operand" or something similar (for the entire code block).

**Review by kflorianski-ibm — CHANGES_REQUESTED** [review]
/meow

File: api/v1alpha1/ibmlicensing_types.go (hunk 1 above)
Comment (Pejdzor) [reply api/v1alpha1/ibmlicensing_types.go:193]: Sounds, better, thanks a lot

Comment (Pejdzor) [reply api/v1alpha1/ibmlicensing_types.go:193]: And appeared

**Review by kflorianski-ibm — APPROVED** [review]

**Pull request merged**